src/topology/running.rs

```diff
@@ -321,18 +329,24 @@ impl RunningTopology {
         // conflicts in their resource usage. So if we combine their
         // resources, all found conflicts are between
         // to be removed and to be added components.
-        let remove_sink = diff
-            .sinks
-            .removed_and_changed()
-            .map(|name| (name, self.config.sinks[name].resources(name)));
-        let add_source = diff
-            .sources
-            .changed_and_added()
-            .map(|name| (name, new_config.sources[name].inner.resources()));
-        let add_sink = diff
-            .sinks
-            .changed_and_added()
-            .map(|name| (name, new_config.sinks[name].resources(name)));
+        let remove_sink = diff.sinks.removed_and_changed().filter_map(|id| {
+            self.config
+                .sinks
+                .get(id)
+                .map(|sink| (id, sink.inner.resources(&id.name)))
+        });
+        let add_source = diff.sources.changed_and_added().filter_map(|id| {
+            new_config
+                .sources
+                .get(id)
+                .map(|source| (id, source.inner.resources()))
+        });
+        let add_sink = diff.sinks.changed_and_added().filter_map(|id| {
+            new_config
+                .sinks
+                .get(id)
+                .map(|sink| (id, sink.inner.resources(&id.name)))
+        });
```

Comment:
This looks like it could be a change in behavior (i.e. `map` to `filter_map` and indexing to `get`). If so, is that intentional?

Reply:
The `indexing` to `get` is due to the fact that we cannot use a struct for indexing (or I don't know about it and I'll look for it).
The `map` to `filter_map` is due to the `get` returning an `Option` but the behavior should be the same in the end.

Reply:
Ok, works with indexing with a struct, don't know why I changed that...